extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamInfoItemExtractor.java

```diff
@@ -62,21 +62,16 @@ public StreamType getStreamType() {
         }
 
         final JsonArray badges = videoInfo.getArray("badges");
-        for (final Object badge : badges) {
-            if (((JsonObject) badge).getObject("metadataBadgeRenderer")
-                    .getString("label", EMPTY_STRING).equals("LIVE NOW")) {
+        for (Object badge : badges) {
+            if (((JsonObject) badge).getObject("metadataBadgeRenderer").getString("label", EMPTY_STRING).equals("LIVE NOW")) {
                 return cachedStreamType = StreamType.LIVE_STREAM;
             }
         }
 
-        final JsonArray thumbnailOverlays = videoInfo.getArray("thumbnailOverlays");
-        for (final Object object : thumbnailOverlays) {
```

Comment:
Maybe it was not a bad idea to cycle through all thumbnail overlays. I'd keep the loop (but obviously revert to the previous condition).